Highway.cpp

```diff
@@ -13,16 +18,32 @@ void Highway::changeSpeed(int newSpeed)
 
 void Highway::addVehicleInternal(Vehicle* v)
 {
-    assert(false);
+    //assert(false);
+
+    if (auto* semiTruck = dynamic_cast<SemiTruck*>(v))
+        semiTruck->blowHorn();
 
+    if (auto* car = dynamic_cast<Car*>(v))
+        car->closeWindows();
+
+    if (auto* motorcycle = dynamic_cast<Motorcycle*>(v))
+        motorcycle->lanesplitAndRace(100);
```

Comment:
use `else if`
If a semi truck is passed in, you're still checking if it is a car or motorcycle
- [ ] revise